### PR TITLE
Subscribe and publish

### PR DESCRIPTION
**Pull request by Nyarish**

### CODE REVIEW

File: ros/src/waypoint_updater/waypoint_updater.py [hunk 1]
```diff
@@ -21,36 +25,164 @@
...
 '''
 
-LOOKAHEAD_WPS = 200 # Number of waypoints we will publish. You can change this number
+# Set Global Variables to use
+
+LOOKAHEAD_WPS = 50 # Number of waypoints we will publish. You can change this number
+LOOKAHEAD_WPS_FOR_STOPPING = 50
+MAX_SPEED = 11.111 # A predefined speed of 11.1111 m/s from waypoint to waypoint
+STOPLINE_BUFFER = 2. # Car stops 2m infront of the stop line.
+DISTANCE_TOO_CLOSE_FOR_STOPPING = 10.# If the car is at full speed and less then this distance ahead of the stop line
```
Comment (shahid-n) [inline ros/src/waypoint_updater/waypoint_updater.py:34]: This is just a test comment.

Global settings look okay for now.

File: ros/src/waypoint_updater/waypoint_updater.py [hunk 2]
```diff
@@ -21,36 +25,164 @@
...
+        self.sub_tl_idx = rospy.Subscriber('/traffic_waypoints', Int32, self.traffic_cb)
+        self.final_waypoints_pub = rospy.Publisher('/final_waypoints', Lane, queue_size=1)
 
         # TODO: Add other member variables you need below
 
-        rospy.spin()
+        # Use spline for the velocity deceleration
+        x = np.array([-0.1, 0., .1, .4, 1., 1.1])
+        y = np.array([0., 0., .1, .5, 1., 1.])
+        self.vel_spline = interpolate.splrep(x, y, s=0)
```
Comment (shahid-n) [inline ros/src/waypoint_updater/waypoint_updater.py:64]: How were these coefficients determined? Is it suggested by the Udacity videos or official project hints?

We might need to figure out a strategy to start testing these kinds of details.

Comment (Nyarish) [reply ros/src/waypoint_updater/waypoint_updater.py:64]: https://classroom.udacity.com/nanodegrees/nd013/parts/01a340a5-39b5-4202-9f89-d96de8cf17be/modules/1dc566d7-03d4-40da-af2c-b8ec85f2e4dd/lessons/407a2efa-3383-480f-9266-5981440b09b3/concepts/3bdfeb8c-8dd6-49a7-9d08-beff6703792d

start from min 10:00

File: ros/src/waypoint_updater/waypoint_updater.py [hunk 3]
```diff
@@ -21,36 +25,164 @@
...
 
         rospy.Subscriber('/current_pose', PoseStamped, self.pose_cb)
         rospy.Subscriber('/base_waypoints', Lane, self.waypoints_cb)
+        rospy.Subscriber('/current_velocity', TwistStamped, self.car_curr_vel_cb)
 
         # TODO: Add a subscriber for /traffic_waypoint and /obstacle_waypoint below
 
-
-        self.final_waypoints_pub = rospy.Publisher('final_waypoints', Lane, queue_size=1)
+        self.sub_tl_idx = rospy.Subscriber('/traffic_waypoints', Int32, self.traffic_cb)
```
Comment (shahid-n) [inline ros/src/waypoint_updater/waypoint_updater.py:56]: Do we want to implement this right from the outset? Until we work on the traffic light detection part, wouldn't it be better to simply ignore traffic lights (so comment this out for now)?

**Comment by Nyarish** [comment]
No we don’t at this point. That’s an error on my part.

**Pull request closed without merging**